### DISCUSSION TITLE
Have we verified if the Source textures are sRGB?

### DISCUSSION THREAD

**Discussion by yogwoggf** [Derpius/VisTrace#97] [Ideas]
As you know, throughput in most path tracing algorithms is represented as a simple vector. The vector represents attenuation from all of the bounces, but it also is a valid linear RGB color. Because it's linear, all the color-related inputs should be linear or else the render can look off. I personally experienced this with Puffy PT and had to run each source texture through a sRGB to linear pass to make sure the lighting is accurate to the albedo of the surfaces that a path bounces off of.

This got me wondering, and I rendered CSS de_dust2 which, in my opinion, is a perfect example of this issue.

**VisTrace**
![vistrace example](https://i.imgur.com/h9FyAcm.png)
**GMod**
![gmod example](https://i.imgur.com/OrbVyXb.jpeg)

And then I wanted to confirm if this was because my vistracer has a linear -> sRGB pass, which could be the cause.
**VisTrace (without linear -> sRGB)**
![vistrace without linear to srgb](https://i.imgur.com/1afTaHN.png)
**GMod**
![gmod example](https://i.imgur.com/OrbVyXb.jpeg)
Also, the albedo output is correct because it's sRGB which is a major 🚩.
![vistrace albedo](https://i.imgur.com/k9CcHG5.png)
From my testing, that should be darker if it's actually linear.

And then, I had Spanky who developed his tracer completely separate from mine test it out.
**VisTrace (Spanky)**
![vistrace spanky](https://cdn.discordapp.com/attachments/826587447236100137/1092626580544704573/render2023-4-3_21-47-23.png)
**GMod (Spanky)**
![gmod spanky](https://cdn.discordapp.com/attachments/826587447236100137/1092626807917908119/image.png)

While his scene setup is.. arguably questionable and doesn't illuminate the scene well, you can still see the color washed out.

VisTrace (both versions) has substantial color changes compared to GMod, and contains a very washed out map. Again, I experienced this in Puffy PT, and I fixed this by.. again, applying a sRGB to linear pass on all of the textures. I did this to VisTrace when the `TraceResult` class [calculates albedo](https://github.com/Derpius/VisTrace/blob/3c0ef660a6644ed1351879ca168c97010e8b9b4f/source/objects/TraceResult.cpp#L228) and got this result.

**VisTrace (patch)**
![vistrace patched](https://i.imgur.com/Un9QkPK.png)
**GMod**
![gmod example](https://i.imgur.com/OrbVyXb.jpeg)

As you can see, the color is much closer to GMod after making all calculations use the linear RGB of the texture.

# Summary
Basically, in a vistracer with just tonemapping and no sRGB encoding, the colors are washed out and my conclusion is that VisTrace is doing linear calculations on sRGB textures.

Anyways, this is a discussion because this could totally be my entire fault and a edge case specific to my vistracer alone, but I wanted to confirm and just discuss it. @Derpius

**Comment by Derpius** [comment]
While the evidence for the claim isn’t great (completely different lighting models and BSDFs), it’s probably the case.

Can you raise an issue on VTFParser to pre-filter images with the sRGB flag? (I’m fairly sure that’s a flag)

**Reply by yogwoggf** [reply]
Sure, also I found more evidence of this.
![evidence](https://i.imgur.com/GI3OgmJ.png)
The `SRGB` flag means that the sRGB curve was already applied to the texture pixels according to the dev wiki. This texture is a random brick wall texture from HL2.
